### ISSUE TITLE
Support generic base machines in PSharp compiler

### ISSUE THREAD

**Issue by muwaqar** [p-org/PSharp#304]
Currently, if in the *.psharp file, if I inherit from a machine that takes generic types, I get a compiler error. This forces me to write all the state machine specification (events and transitions) by hand in CSharp file which is very cumbersome. Another workaround is to create a wrapper/proxy class that inherits from the requisite base class specifying all the generics, and have the real state machine class derive from the proxy class. But that is not very clean.

I try to do (in psharp file):
`internal partial machine ZoneProvisioningStateMachine : Test.NameSpace.StateMachineBase<object, object>`

The error I get:
`
#error Psharp Compiler Error 
 /* 
 Expected base machine identifier.
In TestStateMachine.psharp (line 3): 
internal partial machine ZoneProvisioningStateMachine : Test.NameSpace.StateMachineBase<object, object>
_______________________________________________________________________________________^ 
 */ 
`

**Comment by akashlal** [comment]
I've coded a fix. It is available in PR #305.  We'll merge into master soon but feel free to use that branch if it unblocks your progress.

**Comment by muwaqar** [comment]
@akashlal : How can I consume this? Will a new nuget be released?

**Comment by akashlal** [comment]
@muwaqar I'm preparing a new release today. Its been a while since our last one. It'll also resolve the other issue that you reported. If you find yourself running into further issues, perhaps it'll be better to take a source dependence on the master branch if you can.

**Comment by akashlal** [comment]
@muwaqar Uploading a package to nuget.org is going to be delayed (new MS rules). We have set up an internal feed in the meantime. If you want the nuget, email me on my MS account: akashl
